go/libraries/doltcore/sqle/database_provider.go

```diff
@@ -499,7 +499,7 @@ func (p *DoltDatabaseProvider) CloneDatabaseFromRemote(
 		return err
 	}
 
-	return ConfigureReplicationDatabaseHook(ctx, p, dbName, dEnv)
```

Comment:
My read of this was that it was an error that we called `ConfigureReplicationDatabaseHook` directly here. We had already called `InitDatabaseHook`, which was responsible for this (and possible more hook configuration) on the newly created database. Since `registerNewDatabase()` also calls `InitDatabaseHook`, and there were direct calls to `ConfigureReplicationDatabaseHook` in the `CreateDatabase` path, removing this seemed correct to me.